lib/ingestors/concerns/github_ingestor_material_helpers.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+module Ingestors
+  module Concerns
+    # All methods for the Github Ingestor Class
+    # This involves, the caching, the change from github.com to api.github.com
+    module GithubIngestorMaterialHelpers
+      private
+
+      def resolve_url(repo_data)
+        homepage_nil_or_empty = repo_data['homepage'].nil? || repo_data['homepage'].empty?
+        url = homepage_nil_or_empty ? repo_data['html_url'] : get_redirected_url(repo_data['homepage'])
+        [url, homepage_nil_or_empty]
+      end
+
+      def fetch_homepage_doc(url)
+        response = HTTParty.get(url, follow_redirects: true, headers: { 'User-Agent' => config[:user_agent] })
+        Nokogiri::HTML(response.body)
+      end
+
+      # DEFINITION – Opens the GitHub homepage, fetches the 3 first >25 char <p> tags'text
+      # and joins them with a 'Read more...' link at the end of the description
+      # Some of the first <p> tags were not descriptive, thus skipping them
+      def fetch_definition(doc, url)
+        desc = ''
+        round = 3
+        doc.css('p').each do |p|
+          p_txt = p&.text&.strip&.gsub(/\s+/, ' ')
+          next if (p_txt.length < 25) || round.zero?
+
+          desc = "#{desc}\n#{p_txt}"
+          round -= 1
+        end
+        "#{desc}\n(...) [Read more...](#{url})"
+      end
+
+      # LICENCE – Get proper licence
+      # the licence must match the format of config/dictionaries/licences.yml
+      def fetch_licence(licence)
+        return 'notspecified' if licence.nil? || licence == 'null'
+        return 'other-at' if licence['key'] == 'other'
+
+        licence['spdx_id']
+      end
+
+      # DOI – Fetches DOI from various sources in a repo
+      # I chose to only read the `README.md` as it seems to have the DOI badge almost everytime.
+      # Whereas enabling the fetching of CITATION.cff or CITATION.md would result in increasing
+      # the number of api request.
+      def fetch_doi(full_name)
+        fetch_doi_from_file(full_name, 'README.md')
+
+        # doi = fetch_doi_from_file(full_name, 'CITATION.cff')
+        # return doi if doi
+
+        # doi = fetch_doi_from_file(full_name, 'CITATION.md')
+        # return doi if doi
+      rescue StandardError
+        nil
+      end
```

Comment:
Generally not a good idea to rescue StandardError like this, it will obscure legitimate bugs: typos in method calls etc.